tools/cell_census_builder/experiment_builder.py

```diff
@@ -203,22 +168,15 @@ def accumulate_axes(self, dataset: Dataset, ad: anndata.AnnData, progress: Tuple
         add_tissue_mapping(obs_df, dataset.dataset_id)
 
         # Accumulate aggregation counts
-        self._accumulate_summary_cell_counts(obs_df)
+        self.census_summary_cell_counts = accumulate_summary_counts(self.census_summary_cell_counts, obs_df)
 
         # drop columns we don't want to write
         obs_df = obs_df[list(CENSUS_OBS_TERM_COLUMNS)]
         obs_df = anndata_ordered_bool_issue_853_workaround(obs_df)
 
-        se = soma.Experiment(self.se_uri, context=SOMA_TileDB_Context())
-        assert se.exists()
+        self.populate_obs_axis(obs_df)
 
-        pa_table = pa.Table.from_pandas(
-            obs_df,
-            preserve_index=False,
-            columns=list(CENSUS_OBS_TERM_COLUMNS),
-        )
-        for pa_batch in pa_table.to_batches():
-            se.obs.write(pa_batch)
+        self.dataset_obs_joinid_start[dataset.dataset_id] = self.n_obs
```

Comment:
Now accumulating this data "as we go" instead of re-reading the census obs dataframe later to acquire it; this seemed like a safe replacement